src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -155,13 +157,23 @@ def hook_property(property_name: str) -> Any:
         """
 
         def property_new_share_tensor_getter(_self: "ReplicatedSharedTensor") -> Any:
-            tensor = getattr(_self.tensor, property_name)
-            res = ReplicatedSharedTensor(session=_self.session)
-            res.tensor = tensor
+            tensor1 = getattr(_self.shares[0].tensor, property_name)
+            tensor2 = getattr(_self.shares[1].tensor, property_name)
+            share1 = ShareTensor(session=_self.session)
```

Comment:
RSTensor attribute shares hold a list of torch tensors. The attributes and methods will act upon these torch tensors, not sharetensors. 
Also, parties can hold multiple shares and not necessarily two shares, so it must act on all tensors in the list.

Reply:
It actually applies to all the tensors.Given a secret x  ,each party holds two shares. Applying the hook method or property to an MPCTensor iterates through all the RSTensor and applied the function. Correct me if I am wrong.